image_client/csv_purger.py

```diff
@@ -0,0 +1,113 @@
+import picturae_config as picturae_config
+from importer import Importer
+import traceback
+import pandas as pd
+
+
+class CsvDatabase(Importer):
+    def __init__(self, number):
+        super().__init__(picturae_config, "Botany")
+        self.purge_code = number
+
+    def sql_time_purger(self, database, table, timestamp1, timestamp2):
+        try:
+            cursor = self.specify_db_connection.get_cursor()
+        except Exception as e:
+            self.logger.error(f"Connection Error: {e}")
+
+        sql = f'''DELETE FROM {database}.{table} WHERE TimestampCreated >= 
+                  "{timestamp1}" AND TimestampCreated <= "{timestamp2}"'''
+        self.logger.info(f'running query: {sql}')
+        print(sql)
+        self.logger.debug(sql)
+        try:
+            cursor.execute(sql)
+        except Exception as e:
+            print(f"Exception thrown while processing sql: {sql}\n{e}\n", flush=True)
+            self.logger.error(traceback.format_exc())
+
+        self.specify_db_connection.commit()
+
+        cursor.close()
+
+    def sql_tree_purger(self, database, timestamp1, timestamp2):
+
+        try:
+            cursor = self.specify_db_connection.get_cursor()
+        except Exception as e:
+            self.logger.error(f"Connection Error: {e}")
+        sql_temp = f'''CREATE TEMPORARY TABLE temp_leaf_nodes AS SELECT TaxonID FROM {database}.taxon WHERE TaxonID
+                       NOT IN (SELECT DISTINCT ParentID FROM {database}.taxon
+                       WHERE ParentID IS NOT NULL) 
+                       AND TimestampCreated >= "{timestamp1}" AND TimestampCreated <= "{timestamp2}";'''
+
+        sql_del = f'''DELETE FROM {database}.taxon WHERE TaxonID IN (SELECT TaxonID FROM temp_leaf_nodes);'''
+
+        sql_drop = f'''DROP TEMPORARY TABLE IF EXISTS temp_leaf_nodes;'''
+        rows_affected = 0
+        while True:
+            try:
+                cursor.execute(sql_temp)
+                print(sql_temp)
+                cursor.execute(sql_del)
+                print(sql_del)
+                rows_affected = cursor.rowcount
+                cursor.execute(sql_drop)
+                print(sql_drop)
+            except Exception as e:
+                print(f"Exception thrown while processing sql: \n{e}\n", flush=True)
+                self.logger.error(traceback.format_exc())
+
+            if rows_affected == 0:
+                break
+        self.specify_db_connection.commit()
+
+        cursor.close()
+
```

Comment:
comment ehre (etc - I'll stop these now, just each of these funcs should have a comment block. They don't need to be extensive, just enbough so that when I read them I know what you're trying to do.)